### PR TITLE
feat(dedicated.netapp): add vrack field to netapp network configuration

### PR DESCRIPTION
**Pull request by qpavy**
| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/efs-vrack-services`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12880
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

### CODE REVIEW

File: packages/manager/modules/netapp/src/dashboard/network-configuration/template.html [hunk 1]
```diff
@@ -100,6 +163,6 @@
             <span
                 data-translate="netapp_network_configuration_configure_button"
             ></span>
-        </oui-button>
-    </form>
+        </oui-button> -->
+    <!-- </form> -->
```
Comment (JacquesLarique) [inline packages/manager/modules/netapp/src/dashboard/network-configuration/template.html:167]: Is there still a need for this ?

File: packages/manager/modules/netapp/src/dashboard/network-configuration/service.js [hunk 2]
```diff
@@ -19,10 +34,42 @@ export default class NetAppNetworkConfigurationService {
...
+        this.Apiv2Service.httpApiv2({
+          method: 'put',
+          url: `/engine/api/v2/vrack-services/resource/${vs.id}`,
+          data: vs,
+        }).then(({ data }) => {
+          return {
+            ...vrackAssociationStatus,
+            data,
+          };
+        }),
```
Comment (JacquesLarique) [inline packages/manager/modules/netapp/src/dashboard/network-configuration/service.js:72]:
```suggestion
        }).then(({ data }) => ({
            ...vrackAssociationStatus,
            data,
          })),
```

File: packages/manager/modules/netapp/src/dashboard/network-configuration/routing.js [hunk 3]
```diff
@@ -34,6 +34,15 @@ export default /* @ngInject */ ($stateProvider) => {
...
         return $state.go('^');
       },
+      vracks: /* @ngInject */ (NetAppDashboardService) =>
+        NetAppDashboardService.getVracks().then(({ data }) =>
+          data.map((vrack) => {
+            return {
+              ...vrack,
+              internalName: vrack.iam.urn.split(':').pop(),
+            };
+          }),
```
Comment (JacquesLarique) [inline packages/manager/modules/netapp/src/dashboard/network-configuration/routing.js:44]:
```suggestion
          data.map((vrack) => ({
              ...vrack,
              internalName: vrack.iam.urn.split(':').pop(),
            })),
```

File: packages/manager/modules/netapp/src/dashboard/network-configuration/template.html (hunk 1 above)
Comment (qpavy) [reply packages/manager/modules/netapp/src/dashboard/network-configuration/template.html:167]: nop :)

**Review by JacquesLarique — APPROVED** [review]

File: packages/manager/modules/netapp/src/dashboard/network-configuration/controller.js [hunk 4]
```diff
@@ -50,18 +90,37 @@ export default class OvhManagerNetAppNetworkConfigurationCtrl {
...
         this.goBack().then(() => {
           this.Alerter.success(
             this.$translate.instant(
               'netapp_network_configuration_success_message',
             ),
           );
+
+          if (
+            data.vrackAssociationStatus &&
+            !data.vrackAssociationStatus.success
```
Comment (darsene) [inline packages/manager/modules/netapp/src/dashboard/network-configuration/controller.js:109]: so if `data.vrackAssociationStatus` is null no need to display any alert ?

File: packages/manager/modules/netapp/src/dashboard/network-configuration/template.html [hunk 5]
```diff
@@ -15,91 +15,135 @@
...
+                            ></span>
+                        </button>
+                    </small>
+                </div>
+            </oui-field>
+        </oui-step-form>
+
+        <!-- VRACK SERVICES & SUBNET STEP-->
+        <oui-step-form
+            data-ng-init="stepVrackServices = $ctrl.stepper.vrackServicesSelection"
```
Comment (darsene) [inline packages/manager/modules/netapp/src/dashboard/network-configuration/template.html:63]: Why do you set `stepVrackServices`  property ?

File: packages/manager/modules/netapp/src/dashboard/network-configuration/template.html [hunk 6]
```diff
@@ -15,91 +15,135 @@
...
-                        data-track-name="{{:: $ctrl.TRACKING_BASE + '::add-vrack-services'}}"
-                        data-track-type="action"
-                    >
+    <oui-stepper
+        data-on-finish="$ctrl.configureNetwork()"
+        data-current-index="$ctrl.currentStep"
+    >
+        <!-- VRACK STEP -->
+        <oui-step-form
+            data-ng-init="stepVrack = $ctrl.stepper.vrackSelection"
```
Comment (darsene) [inline packages/manager/modules/netapp/src/dashboard/network-configuration/template.html:24]: Why do you set `stepVrack` property ?

File: packages/manager/modules/netapp/src/dashboard/network-configuration/controller.js (hunk 4 above)
Comment (qpavy) [reply packages/manager/modules/netapp/src/dashboard/network-configuration/controller.js:109]: I would say yes but I am still not sure :)

**Review by darsene — APPROVED** [review]

**Pull request merged**